src/charm.py

```diff
@@ -561,57 +636,80 @@ def _update_kratos_endpoints_relation_data(self, event: RelationEvent) -> None:
         )
         self.endpoints_provider.send_endpoint_relation_data(admin_endpoint, public_endpoint)
 
+    def _patch_statefulset(self) -> None:
+        if not self.unit.is_leader():
+            return
+
+        pod_spec_patch = {
+            "containers": [
+                {
+                    "name": self._container_name,
+                    "volumeMounts": [
+                        {
+                            "mountPath": str(self._config_dir_path),
+                            "name": "config",
+                            "readOnly": True,
+                        },
+                    ],
+                },
+            ],
+            "volumes": [
+                {
+                    "name": "config",
+                    "configMap": {"name": self._kratos_config_map_name},
+                },
+            ],
+        }
+        patch = {"spec": {"template": {"spec": pod_spec_patch}}}
+        self.client.patch(StatefulSet, name=self.meta.name, namespace=self.model.name, obj=patch)
```

Comment:
Have a question about the `client` here. I saw this `client` is also used in the `ConfigMapHandler` class.

I think it would be better to isolate the client of a third-party library into an individual client class. In this way, the changes in the third-party APIs will not infiltrate different places, causing changes in multiple places.